src/condor_schedd.V6/schedd.cpp

```diff
@@ -17418,25 +17426,15 @@ Scheduler::launch_local_startd() {
 	}
 
 	std::string daemon_sock = SharedPortEndpoint::GenerateEndpointName( "local_universe_startd" );
-	m_local_startd_pid = daemonCore->Create_Process(	path.c_str(),
-										args,
-										PRIV_ROOT,
-										rid, 
-	                                  	1, /* is_dc */
-	                                  	1, /* is_dc */
-										&env, 
-										NULL, 
-										NULL,
-										NULL, 
-	                                  	NULL,  /* stdin/stdout/stderr */
-										NULL, 
-										0,    /* niceness */
-									  	NULL,
-										create_process_opts,
-										NULL,
-										NULL,
-										daemon_sock.c_str() );
 
+    std::string create_process_err_msg;
+	OptionalCreateProcessArgs cpArgs(create_process_err_msg);
+	m_local_startd_pid = daemonCore->CreateProcessNew( path, args,
+		 cpArgs.priv(PRIV_ROOT)
+		 .reaperID(rid)
+		 .wantCommandPort(true).wantUDPCommandPort(true)
+		 .daemonSock(daemon_sock.c_str())
+	);
```

Comment:
wantCommandPort() and wantUDPCommandPort() can be omitted, since these are the default values. You need to add an env() call.